lib/resqued/listener.rb

```diff
@@ -63,12 +63,13 @@ def self.exec!
     end
 
     SIGNALS = [:CONT, :QUIT, :INT, :TERM].freeze
-    ALL_SIGNALS = SIGNALS + [:CHLD]
+    ALL_SIGNALS = SIGNALS + [:CHLD, :HUP]
```

Comment:
do we need add `HUP` here since we already handle HUP through `trap(:HUP) {} ` in both listener and worker process?

Reply:
This is here to help the code be more self-documenting and complete. `ALL_SIGNALS` is a list of all of the signals the listener installs handlers for. I thought about just leaving the HUP handler in place so that it doesn't need to redefine it, but didn't end up with a clean way both to list all of the listener's signals and to have the worker not remove it. So I opted to list it here and then define the handler in the worker as well. Doing it this way also makes it easier for the handlers to diverge in the future.